src/renderer/routes/test_sequencer_panel/utils/SequenceHandler.ts

```diff
@@ -0,0 +1,356 @@
+    });
+    return new Ok(undefined);
+  } catch (e: unknown) {
+    return buildResultFromCatch(e);
+  }
+}
+
+export async function importSequence(
+  filePath: string,
+  fileContent: string,
+  stateManager: StateManager,
+  skipImportDeps: boolean = false,
+): Promise<Result<void, Error>> {
+  // From a file, import a sequence and update the test sequencer
+  // * Importing a sequence overwrites the current sequence, even the test sequencer is unsaved
+  try {
+    let sequence = readJsonSequence(fileContent);
+    if (!sequence) {
+      throw new Error("Error reading sequence file");
+    }
+    const sequencePath = filePath.replaceAll(sequence.name + ".tjoy", "");
+    sequence = updatePath(sequence, sequencePath);
+    if (!skipImportDeps) {
+      const success = await installDeps(sequence);
+      if (!success) {
+        throw Error("Not able to installing dependencies");
+      }
+    }
+    await syncSequence(sequence, stateManager);
+    return new Ok(undefined);
+  } catch (e: unknown) {
+    return buildResultFromCatch(e);
+  }
+}
+
+export async function exportSequence(
+  stateManager: StateManager,
+): Promise<Result<void, Error>> {
+  // Export the current sequence as a zip file without any dependencies
+  await saveSequence(stateManager);
+  const error = new Error("Export Not Implemented");
+  return new Err(error);
+}
+
+export async function closeSequence(
+  stateManager: StateManager,
+): Promise<Result<void, Error>> {
+  // Close the current proejct from the app. The sequence is NOT deleted from disk
+  if (stateManager.project !== null) {
+    stateManager.removeSequence(stateManager.project.name);
+  }
+  return new Ok(undefined);
+}
+
+export async function verifyElementCompatibleWithSequence(
+  sequence: TestSequencerProject,
+  elements: TestSequenceElement[],
+): Promise<Result<void, Error>> {
+  // Verify that the elements are within the current sequence directory.
+  try {
+    await throwIfNotInAllBaseFolder(elements, sequence.projectPath);
+    return new Ok(undefined);
+  } catch (e: unknown) {
+    return buildResultFromCatch(e);
+  }
+}
+
+// Private -------------------------------------------------------------------------------------------------
+
+function buildResultFromCatch(e: unknown): Result<void, Error> {
+  if (e instanceof Error) {
+    return new Err(e);
+  } else {
+    console.error("[Save Sequence] Unknown error", e);
+    return new Err(new Error("Unknown error while creating the sequence."));
+  }
+}
+
+function validatePath(sequence: TestSequencerProject): TestSequencerProject {
+  let path = sequence.projectPath;
+  if (!path.endsWith("/")) {
+    path = path + "/";
+  }
+  return updatePath(sequence, path);
+}
+
+function updatePath(
+  sequence: TestSequencerProject,
+  newPath: string,
+): TestSequencerProject {
+  return {
+    ...sequence,
+    projectPath: newPath,
+  };
+}
+
+async function saveToDisk(sequence: TestSequencerProject): Promise<void> {
+  if ("api" in window) {
```

Comment:
you don't need to check if api is in window everywhere, we're always running within the electron context